build.gradle

```diff
@@ -68,40 +60,37 @@ ensureBuildPrerequisites(buildPrerequisitesMessage)
 final htsjdkVersion = System.getProperty('htsjdk.version', '3.0.0')
 final googleNio = 'com.google.cloud:google-cloud-nio:0.123.25'
 
-// Get the jdk files we need to run javaDoc. We need to use these during compile, testCompile,
-// test execution, and gatkDoc generation, but we don't want them as part of the runtime
-// classpath and we don't want to redistribute them in the uber jar.
-final javadocJDKFiles = ToolProvider.getSystemToolClassLoader() == null ? files([]) : files(((URLClassLoader) ToolProvider.getSystemToolClassLoader()).getURLs())
-
 configurations {
     cloudConfiguration {
-        extendsFrom runtime
+        extendsFrom implementation
         dependencies {
             cloudConfiguration(googleNio)
         }
     }
 }
 
 dependencies {
-    compile('com.intel.gkl:gkl:0.8.8') {
+    implementation('com.intel.gkl:gkl:0.8.8') {
         exclude module: 'htsjdk'
     }
-    compile 'com.google.guava:guava:15.0'
-    compile 'org.apache.commons:commons-math3:3.5'
-    compile 'org.apache.commons:commons-collections4:4.3'
-    compile 'commons-lang:commons-lang:2.6'
-    compile 'com.github.samtools:htsjdk:' + htsjdkVersion
-    compile 'org.broadinstitute:barclay:4.0.2'
-    compile 'org.apache.logging.log4j:log4j-api:2.17.1'
-    compile 'org.apache.logging.log4j:log4j-core:2.17.1'
+    implementation 'com.google.guava:guava:15.0'
+    implementation 'org.apache.commons:commons-math3:3.5'
+    implementation 'org.apache.commons:commons-collections4:4.3'
+    implementation 'commons-lang:commons-lang:2.6'
+    implementation 'com.github.samtools:htsjdk:' + htsjdkVersion
+    implementation 'org.broadinstitute:barclay:4.0.2'
+    implementation 'org.apache.logging.log4j:log4j-api:2.17.1'
+    implementation 'org.apache.logging.log4j:log4j-core:2.17.1'
+
+    // replacement for jdk nashorn engine GPL-2.0 license
+    implementation 'org.openjdk.nashorn:nashorn-core:15.4'
+
+    //NOTE: this dependency spec causes doc gen to fail on FingerPrintChecker since doc gen is not
+    // a compile time task
     compileOnly(googleNio)
```

Comment:
Do we know what configuraton javadoc uses by default?  I'm sure we can set the javadoc task to use the compile configuration if necessary.